scene/2d/camera_2d.cpp

```diff
@@ -75,6 +75,22 @@ void Camera2D::_update_process_mode() {
 	}
 }
 
+void Camera2D::_setup_viewport() {
+	if (custom_viewport && ObjectDB::get_instance(custom_viewport_id)) {
+		viewport = custom_viewport;
+	} else {
+		viewport = get_viewport();
+	}
+
+	RID vp = viewport->get_viewport_rid();
+	group_name = "__cameras_" + itos(vp.get_id());
+	canvas_group_name = "__cameras_c" + itos(canvas.get_id());
+	add_to_group(group_name);
+	add_to_group(canvas_group_name);
+
+	viewport->connect("size_changed", this, "_update_scroll");
```

Comment:
Note that this will call `_update_scroll()` on viewport resize, regardless of whether we have smoothing enabled or not (and with smoothing on the update will also be done by processing). But I assumed it's not a big deal for this situation.

Reply:
I think it should be fine even if called twice in this situation. :+1: